File: backend/app/agents/prompts/assessment.py
ASSESSMENT_SYSTEM_PROMPT = """You are an expert career counselor and learning path designer. Your role is to assess a user's current skills, experience, and goals to create a personalized learning roadmap.

Your task is to analyze the user's responses and create a comprehensive assessment that will be used to generate a tailored learning roadmap.

You must output a valid JSON object with the following structure:

{
  "goal": {
    "title": "The user's learning goal (e.g., 'AWS Solutions Architect')",
    "description": "Detailed description of what the user wants to achieve",
    "timeframe": "Target timeframe in weeks (e.g., 16 for 4 months)",
    "motivation": "User's motivation and why they want to achieve this goal"
  },
  "currentSkills": {
    "technical": ["Array of technical skills user currently has"],
    "soft": ["Array of soft skills user currently has"],
    "experience": {
      "years": "Number of years of relevant experience",
      "roles": ["Previous or current roles"],
      "projects": ["Notable projects or achievements"]
    }
  },
  "learningPreferences": {
    "pace": "fast | moderate | slow",
    "style": "hands-on | theoretical | balanced",
    "hoursPerWeek": "Number of hours user can dedicate per week",
    "preferredResources": ["video", "articles", "documentation", "courses", "books"]
  },
  "constraints": {
    "availableTime": "Description of time availability",
    "budget": "free | low | moderate | high",
    "prerequisites": ["Any prerequisites user has already completed"],
    "challenges": ["Anticipated challenges or concerns"]
  }
}

Be thorough, ask clarifying questions if needed, and ensure the assessment accurately captures the user's current state and aspirations."""

def build_assessment_prompt(answers: dict) -> str:
    challenges_text = f"**Anticipated Challenges**: {answers.get('challenges', '')}" if answers.get('challenges') else ""
    
    return f"""Based on the following user responses, create a comprehensive assessment:

**Learning Goal**: {answers.get('goal', '')}
**Target Timeframe**: {answers.get('timeframe', '')} weeks
**Motivation**: {answers.get('motivation', '')}
**Current Role**: {answers.get('currentRole', '')}
**Technical Skills**: {answers.get('technicalSkills', '')}
**Years of Experience**: {answers.get('experience', '')} years
**Learning Style**: {answers.get('learningStyle', '')}
**Hours per Week**: {answers.get('hoursPerWeek', '')} hours
**Budget**: {answers.get('budget', '')}
{challenges_text}

Analyze these responses and create a structured assessment that captures:
1. The user's clear goal and what they want to achieve
2. Their current skill level and experience
3. Their learning preferences and constraints
4. Any gaps or areas they need to focus on

Output the assessment as a valid JSON object following the specified schema."""

File: frontend/lib/agents/prompts/assessment.ts
export interface Question {
    id: string
    question: string
    type: 'text' | 'textarea' | 'select'
    options?: { label: string; value: string }[]
    placeholder?: string
    required?: boolean
}

export const ASSESSMENT_QUESTIONS: Question[] = [
    {
        id: 'goal',
        question: 'What is your primary goal for this roadmap?',
        type: 'textarea',
        placeholder: 'e.g., Become a Senior PM, Transition into AI Product Management...',
        required: true,
    },
    {
        id: 'experience',
        question: 'How many years of relevant experience do you have?',
        type: 'select',
        options: [
            { label: '0-2 years', value: '0-2' },
            { label: '3-5 years', value: '3-5' },
            { label: '5-8 years', value: '5-8' },
            { label: '8+ years', value: '8+' },
        ],
        required: true,
    },
    {
        id: 'current_role',
        question: 'What is your current role?',
        type: 'text',
        placeholder: 'e.g., Software Engineer, Marketing Manager, Junior PM',
        required: true,
    },
    {
        id: 'skills',
        question: 'What are your top 3 current skills?',
        type: 'textarea',
        placeholder: 'e.g., Data Analysis, User Research, Python...',
        required: true,
    },
    {
        id: 'learning_style',
        question: 'How do you prefer to learn?',
        type: 'select',
        options: [
            { label: 'Hands-on (Building projects)', value: 'hands-on' },
            { label: 'Theoretical (Reading/Courses)', value: 'theoretical' },
            { label: 'Balanced', value: 'balanced' },
        ],
        required: true,
    },
    {
        id: 'time_commitment',
        question: 'How many hours per week can you dedicate?',
        type: 'select',
        options: [
            { label: '5-10 hours', value: '5-10' },
            { label: '10-20 hours', value: '10-20' },
            { label: '20+ hours', value: '20+' },
        ],
        required: true,
    },
]
